driver/bpf/fillers.h

```diff
@@ -722,14 +757,28 @@ FILLER(sys_readv_preadv_x, true)
 	if (res != PPM_SUCCESS)
 		return res;
 
-	iov = (const struct iovec __user *)bpf_syscall_get_argument(data, 1);
-	iovcnt = bpf_syscall_get_argument(data, 2);
+	/*
+	* data and size
+	*/
+	if (retval > 0)
```

Comment:
Here we need to choose if we want to discard the payload in case of failure, same question we have here  https://github.com/falcosecurity/libs/pull/900/files#r1113393171

Reply:
probably the best approach would be to get all the data also when the syscall fails, we will read junk in most of the cases but we could also catch some useful data in case of strange attack patterns

Reply:
btw in this PR you can keep it as it, we will adapt all the case in a separate one since we have to adapt all possible cases

Reply:
Agreed in doing this in a separate PR